### ISSUE TITLE
refactor: make the CMAPSS example a package and drop the sys.path bootstrap

### ISSUE THREAD

**Issue by debrevitatevitae** [GroupiSP/probreg#24]
## What to build

Turn `examples/jax/cmapss/` into a proper Python package, so that the example's modules and their tests import each other like ordinary Python instead of bootstrapping `sys.path` and hand-loading modules through `importlib`.

Today every module in the example that needs a sibling has to prepend its own directory to `sys.path` before it can import it. `examples/jax/cmapss/run.py:38-43` and `examples/jax/cmapss/plots.py:34-39` carry the *same* block, comment and all:

```python
# Allow sibling-module imports (`data`, `model`, `preprocessing`) both when
# run as a script and when loaded from an arbitrary working directory, e.g.
# via `importlib` in tests.
_MODULE_DIR = Path(__file__).resolve().parent
if str(_MODULE_DIR) not in sys.path:
    sys.path.insert(0, str(_MODULE_DIR))

from data import (...)
```

The imports that follow it are module-level imports after code, so ruff has to be told to look away — `pyproject.toml`, `[tool.ruff.lint.per-file-ignores]`:

```toml
"examples/jax/cmapss/*.py" = ["E402"]
"tests/jax/test_cmapss_*.py" = ["E402"]
```

And because the example is not importable, each of the five test modules re-implements the same module-loading dance —
`tests/jax/test_cmapss_data.py:17-24`, `tests/jax/test_cmapss_model.py:14-21`, `tests/jax/test_cmapss_preprocessing.py:17-28`, `tests/jax/test_cmapss_plots.py:19-28`, `tests/jax/test_cmapss_run.py:21-30` — a `spec_from_file_location` / `module_from_spec` / `sys.modules[...]` / `exec_module` sequence, plus its own `sys.path.insert` where the loaded module in turn imports a sibling. Every test then reaches through an `_MODULE`/`_RUN`/`_PLOTS` handle rather than naming what it imports.

Making the directory a package removes the cause rather than the symptom: `examples/`, `examples/jax/`, and `examples/jax/cmapss/` get `__init__.py`, the modules use package-relative (or absolute `examples.jax.cmapss.*`) imports, all bootstrap blocks are deleted, the tests import normally, and both `E402` ignores go away.

The example is then run as a module. The README currently documents:

```shell
uv run --group example-cmapss python examples/jax/cmapss/run.py
```

which becomes:

```shell
uv run --group example-cmapss python -m examples.jax.cmapss.run
```

The same applies to the other two commands the README gives — `python examples/jax/cmapss/data.py` / `--fetch` (`examples/jax/cmapss/README.md:29-33`), and `run.py --plot-path rul_curves.png` (`README.md:126-129`) — and to the identical invocation lines in the modules' own docstrings (`run.py:21-23`, `data.py:9-17`).

## Why this is a separate ticket

This is deliberately deferred out of the `feat/cmapss-plots` branch (#19, #22). There, the duplicate bootstrap block in `plots.py` is simply deleted as a stopgap: `run.py` remains the single bootstrap, and since `run.py` imports `plots`, the `sys.path` entry it installs is already in place by the time `plots` needs its sibling. That removes the verbatim duplication without touching the run command, the tests, or the ruff configuration — all of which this ticket does in one deliberate move instead of drifting into a plotting feature branch.

## Acceptance criteria

- [ ] `examples/`, `examples/jax/`, and `examples/jax/cmapss/` are importable packages (`__init__.py` present, with whatever the project's other packages carry).
- [ ] No `_MODULE_DIR` / `sys.path.insert` block remains in any module under `examples/jax/cmapss/`, nor in any `tests/jax/test_cmapss_*.py`.
- [ ] Sibling imports in the example use explicit relative or absolute package imports, sit in the normal import block at the top of the file, and are sorted by ruff's isort rules.
- [ ] No `tests/jax/test_cmapss_*.py` uses `importlib.util.spec_from_file_location`; each imports the names it needs directly (`from examples.jax.cmapss.model import Cnn1DMeanModel`, and so on), keeping the existing `pytest.importorskip` guards for optional dependencies.
- [ ] Both `E402` entries are removed from `[tool.ruff.lint.per-file-ignores]` in `pyproject.toml`, and no `# noqa: E402` is introduced to replace them.
- [ ] `examples/jax/cmapss/README.md` and the module docstrings quote the `python -m examples.jax.cmapss.<module>` form for every documented invocation, and the documented commands have been run and work.
- [ ] `python -m examples.jax.cmapss.data --fetch` and `python -m examples.jax.cmapss.run --plot-path <path>` behave exactly as their script forms do today.
- [ ] The full `pytest` suite passes and `pre-commit run --all-files` passes.

## Out of scope

- Any behavioural change to the example: no change to preprocessing, model architecture, training loop, the FD001 test protocol, or the figure. The printed metrics — RMSE, 95% interval coverage, point-CRPS — must be **unchanged for a fixed seed**, and so must the metric names and their order.
- Any change to the `probreg` library itself, or to non-CMAPSS examples and their tests beyond the `__init__.py` files needed to make `examples/` and `examples/jax/` packages.
- Restructuring or renaming the example's modules, splitting `run.py`, or moving the example out of `examples/`.
- Packaging the example for distribution (entry points, inclusion in the wheel, a `[project.scripts]` console script).
- Rewriting the tests' assertions or converting them to a different style; the change is to how they obtain the code under test, not to what they check.
- Anything on the `feat/cmapss-plots` branch. That branch keeps its stopgap; this ticket lands afterwards.

## Blocked by

- #22 — the `feat/cmapss-plots` work should land first, so this refactor rewrites a settled set of modules rather than racing an open branch.
